projects/packages/autoloader/src/class-plugins-handler.php

```diff
@@ -155,12 +146,6 @@ public function cache_plugins( $plugins ) {
 	public function have_plugins_changed( $plugins ) {
 		global $jetpack_autoloader_cached_plugin_paths;
 
-		// When no autoloader has executed there is nothing to have changed.
-		if ( ! isset( $jetpack_autoloader_cached_plugin_paths ) ) {
```

Comment:
Since older autoloaders lack the `$jetpack_autoloader_cached_plugin_paths` global we need to perform a reset when it isn't present. This prevents us from having to do a latest version comparison because we can assume we're either performing a first initialization or that the previous autoloader search did not include the same plugins.